pkg/controller/iam/user/controller.go

```diff
@@ -69,6 +72,9 @@ func SetupUser(mgr ctrl.Manager, o controller.Options) error {
 		Complete(managed.NewReconciler(mgr,
 			resource.ManagedKind(v1beta1.UserGroupVersionKind),
 			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: iam.NewUserClient}),
+			managed.WithInitializers(
+				&externalNamer{kube: mgr.GetClient()},
```

Comment:
The default initializer list includes an external name initializer from runtime, so you can use that one here as is. See an example [here](https://github.com/crossplane/provider-aws/blob/793e588/pkg/controller/ecr/repository/controller.go#L81).

Reply:
ah, thank you, that's so much better, done.